table_schema_cache.go

```diff
@@ -211,6 +211,77 @@ func LoadTables(db *sql.DB, tableFilter TableFilter, columnCompressionConfig Col
 	return tableSchemaCache, nil
 }
 
+func (ts *TableSchema) findColumnByName(name string) *schema.TableColumn {
+	for _, column := range ts.Columns {
+		if column.Name == name {
+			return &column
+		}
+	}
+	return nil
+}
+
+func NonExistingPaginationKeyColumnError(schema, table, paginationKey string) error {
+	return fmt.Errorf("Pagination Key `%s` for %s.%s non existent", paginationKey, schema, table)
+}
+
+func NonExistingPKError(schema, table string) error {
+	return fmt.Errorf("%s.%s has no Primary Key to default to for Pagination purposes. Kindly specify a Pagination Key for this table in the CascadingPaginationColumnConfig", schema, table)
+}
+
+func CompositePKError(schema, table string) error {
+	return fmt.Errorf("%s.%s attempted to default to a Composite Primary Key for Pagination purposes whereas this is not supported. Kindly specify a Pagination Key for this table in the CascadingPaginationColumnConfig", schema, table)
+}
+
+func (ts *TableSchema) paginationKeyColumn(dbname, table string, cascadingPaginationColumnConfig *CascadingPaginationColumnConfig) (*schema.TableColumn, error) {
+	var err error
+	var paginationKeyColumn *schema.TableColumn
+
+	if paginationColumn, found := cascadingPaginationColumnConfig.PaginationColumnFor(dbname, table); found {
+		paginationKeyColumn = ts.findColumnByName(paginationColumn)
+		err = NonExistingPaginationKeyColumnError(dbname, table, paginationColumn)
+	} else {
+		if cascadingPaginationColumnConfig == nil || cascadingPaginationColumnConfig.FallbackColumn == "" {
```

Comment:
Instead of nesting like this, can you just have:

```
if cascadingPAginationColumnConfig == nil {
  paginationKeyColumn = findViaPK
} else if cascadingPaginationColumnConfig.FallbackColumn == "" {
  paginationKEyColumn = findViaFallback
} else if paginationColumn, found := cascadingPaginationColumnConfig.PaginationColumnFor(dbname, table); found {
  ...
} else {
  // do we need this clause or does the above already cover all possibilities?
}
```

Reply:
This cannot be split up as you suggested:
```
if cascadingPAginationColumnConfig == nil {
  paginationKeyColumn = findViaPK
} else if cascadingPaginationColumnConfig.FallbackColumn == "" {
  paginationKEyColumn = findViaFallback
```
as both cases need to `findViaPK`. 
So the following would work:
if cascadingPAginationColumnConfig == nil {
  paginationKeyColumn = findViaPK
} else if cascadingPaginationColumnConfig.FallbackColumn == "" {
  paginationKEyColumn = findViaPK
```
but I am not sure that this increases clarity.

Nevertheless, I can introduce more helper functions if this is deemed important to understand that the checks being done need to reflect the cases as documented at: https://github.com/Shopify/ghostferry/blob/090abdce25b3821280e0503a5b6d26be0f05d423/config.go#L271-L281

Reply:
I meant to write:

```
if cascadingPAginationColumnConfig == nil || cascadingPaginationColumnConfig.FallbackColumn == "" {
  paginationKeyColumn = findViaPK
} else if cascadingPaginationColumnConfig.FallbackColumn != "" {
  paginationKEyColumn = findViaFallbackCol
```

I was just suggesting that we turn this method flat (right now it nested 2 levels deep). If you think it's still not clear, you add some inline comments as well.